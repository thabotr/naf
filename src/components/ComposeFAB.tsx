import { ViewProps } from "react-native";
import { FAB } from "react-native-paper";
import { useTheme } from "../context/theme";

function ComposeFAB({onLongPress, onPress}:{onLongPress:()=>void, onPress:()=>void}){
  const {theme} = useTheme();

  return <FAB
  style={{
    margin: 3,
    borderRadius: 0,
    backgroundColor: theme.color.userSecondary,
  }}
  icon="pencil"
  onLongPress={onLongPress}
  onPress={onPress}
/>
}

export {ComposeFAB};