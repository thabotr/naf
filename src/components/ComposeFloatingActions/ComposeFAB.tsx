import React from 'react';
import {StyleSheet} from 'react-native';
import {FAB} from 'react-native-paper';
import {useTheme} from '../../shared/providers/theme';

function ComposeFAB({
  onLongPress,
  onPress,
}: {
  onLongPress: () => void;
  onPress: () => void;
}) {
  const {theme} = useTheme();
  const styles = StyleSheet.create({
    fab: {
      margin: 3,
      borderRadius: 0,
      backgroundColor: theme.color.userSecondary,
    },
  });
  return (
    <FAB
      style={styles.fab}
      icon="pencil"
      onLongPress={onLongPress}
      onPress={onPress}
    />
  );
}

export {ComposeFAB};
