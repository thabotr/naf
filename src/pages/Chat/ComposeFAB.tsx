import React from 'react';
import {ViewProps} from 'react-native';
import {IconButton} from 'react-native-paper';
import globalStyles from '../../shared/styles';

type Props = ViewProps & any & {};

export default function (props: Props) {
  return (
    <IconButton
      accessibilityLabel="compose message"
      icon="pencil"
      style={[globalStyles.square, props.style]}
      {...props}
    />
  );
}
