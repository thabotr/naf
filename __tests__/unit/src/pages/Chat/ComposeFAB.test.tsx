import React from 'react';
import renderer from 'react-test-renderer';
import ComposeFAB from '../../../../../src/pages/Chat/ComposeFAB';

describe('Chat', () => {
  describe('ComposeFAB', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<ComposeFAB />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
