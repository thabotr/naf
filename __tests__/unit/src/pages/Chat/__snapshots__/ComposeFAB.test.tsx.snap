// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat ComposeFAB renders correctly 1`] = `
<View
  accessibilityLabel="compose message"
  accessibilityRole="button"
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  focusable={false}
  hitSlop={
    {
      "bottom": 6,
      "left": 6,
      "right": 6,
      "top": 6,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "overflow": "hidden",
      },
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 6,
          "overflow": "hidden",
        },
        {
          "borderRadius": 18,
          "height": 36,
          "width": 36,
        },
        undefined,
        [
          {
            "borderRadius": 0,
          },
          undefined,
        ],
      ],
    ]
  }
>
  <View>
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        [
          {
            "color": "#000000",
            "fontSize": 24,
          },
          [
            {
              "lineHeight": 24,
              "transform": [
                {
                  "scaleX": 1,
                },
              ],
            },
            {
              "backgroundColor": "transparent",
            },
          ],
          {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      󰏫
    </Text>
  </View>
</View>
`;
